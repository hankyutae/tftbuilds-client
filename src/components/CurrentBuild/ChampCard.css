.gold {
  color: orange;
}
.silver{
  color:silver;
}
.bronze{
  color:rgb(112, 26, 26);
}


.build-display-champion-card{
  position:relative;
  background-color: azure;
  padding:4px 2px;
  margin:10px 0px;
}
.build-display-champion-card>.edit-notice{
  position:absolute;/* 
  color:rgb(238, 208, 172); */
  color:red;
  top:0;
  left:2px;

}

.champion-simple-flex{
  display:flex;
  align-items:center;
  justify-content:space-around;
}

.build-champ-icon{
  width:50px;
}
.build-display-champion-card p{
  margin:0;
  padding:0;
}

.build-display-champion-card .b1{
  flex: 1 1 25%;
}
.build-display-champion-card .b2{
  flex: 1 1 25%;
}
.build-display-champion-card .b3{
  flex: 1 1 25%;
}
.build-display-champion-card .b4{
  flex: 1 1 140px;
}


.item-img-div{
  display:inline-block;
}

.build-item-icon{
  width:35px;
  margin-right:5px;
}


.remove-champ-button{
  color:red;
}

















.build-display-card-expansion{
  position:relative;
  margin:5px;
  padding:5px;
  padding-top:17px;
  background-color:rgb(225, 243, 202);
  color:black;
  text-align:left;
}
.build-display-card-expansion>section{
  
  margin:5px;
  padding:5px;
  border:1px solid grey;
}
.build-display-stats{
  display:flex;
  flex-direction:row;
  justify-content:center;
  align-items:center;
}
.build-stats-table{
  border: 1px solid black;
  border-collapse: collapse;
  flex-grow:2;
}
.build-stats-table td{
  border: 1px solid black;
  padding:2px;
}


.build-display-card-expansion>.edit-notice{
  position:absolute;
  color:black(238, 208, 172);
  top:0px;
  right:2px;
}


.build-display-expanded-show-items{
  display:flex;
  align-items:center;
  justify-content:center;
}