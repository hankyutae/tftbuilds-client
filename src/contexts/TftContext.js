import React from 'react';

export const TftContext = React.createContext({
  error: null,
  arrayOfPrices:[1,2,3,4,5],
  arrayOfStars:['bronze','silver','gold'],
  expandedTraits:{},
  numOfBasicItems:9,
  loginState:0,
  toggleLoginState :  ()=>{ },
  setError: () => { },
  clearError: () => { },
  setChampions: () => { },
  setItems: () => { },
  setTraits: () => { },
  newCurrentBuild: () => { },
  addChampToCurrentBuild: () => { },
  removeChampFromCurrentBuild: ()=>{},
  changeStarsOnChamp:() =>{},
  lowerStarsOnChamp:() =>{},
  raiseStarsOnChamp: () =>{},
  addItemToChamp:() =>{},
  removeItemFromChamp:() =>{},
  switchItemOnChamp:() =>{},
  toggleExpandedOnCurrentBuild:()=>{},
  toggleStatsOnCurrentBuild:()=>{},
  toggleAddItemOnCurrentBuild:()=>{},
  toggleUltOnCurrentBuild:()=>{},
  toggleTraitDesc:()=>{},
  champions: {},
  items: {},
  traits: {},
  currentBuild: []
});

export default TftContext;

export class TftProvider extends React.Component {
  state = {
    arrayOfPrices:[1,2,3,4,5,7],
    arrayOfStars:['bronze','silver','gold'],
    numOfBasicItems:9,
    champions: this.props.champions||{},
    items: this.props.items||{},
    traits: this.props.traits||{},
    currentBuild:this.props.currentBuild||[],
    expandedTraits:this.props.expandedTraits ||{},
    loginState:0,
    toggleLoginState:()=>{
      console.log('even made it here');
      this.setstate({
        loginState:(this.state.loginState+1)
      })
    },
    toggleTraitDesc:(trait)=>{
      let newTraitList={...this.state.expandedTraits};
      newTraitList[trait]=!newTraitList[trait];
      this.setState({
        expandedTraits:newTraitList
      })
    },
    toggleExpandedOnCurrentBuild:(index)=>{
      let newBuild=[...this.state.currentBuild];
      newBuild[index]={...newBuild[index], isExpanded:!(newBuild[index].isExpanded)};
      this.setState({
        currentBuild: newBuild
      })
    },
    toggleStatsOnCurrentBuild:(index)=>{
      let newBuild=[...this.state.currentBuild];
      newBuild[index]={...newBuild[index], isStatsExpanded:!(newBuild[index].isStatsExpanded)};
      this.setState({
        currentBuild: newBuild
      })
    },
    toggleAddItemOnCurrentBuild:(index)=>{
      let newBuild=[...this.state.currentBuild];
      newBuild[index]={...newBuild[index], isAddItemExpanded:!(newBuild[index].isAddItemExpanded)};
      this.setState({
        currentBuild: newBuild
      })
    },
    toggleUltOnCurrentBuild:(index)=>{
      let newBuild=[...this.state.currentBuild];
      newBuild[index]={...newBuild[index], isUltExpanded:!(newBuild[index].isUltExpanded)};
      this.setState({
        currentBuild: newBuild
      })
    },

    changeStarsOnChamp:(index, stars)=>{
      let newBuild=[...this.state.currentBuild];
      newBuild[index]={...newBuild[index], stars:stars};
      this.setState({
        currentBuild: newBuild
      })
    },
    lowerStarsOnChamp:(index)=>{
      let newBuild=[...this.state.currentBuild];
      newBuild[index]={...newBuild[index], stars:newBuild[index].stars<=1? 1: newBuild[index].stars-1};
      this.setState({
        currentBuild: newBuild
      })
    },
    raiseStarsOnChamp:(index)=>{
      let newBuild=[...this.state.currentBuild];
      newBuild[index]={...newBuild[index], stars:newBuild[index].stars>=this.state.arrayOfStars.length? this.state.arrayOfStars.length: newBuild[index].stars+1};
      this.setState({
        currentBuild: newBuild
      })
    },

    addItemToChamp:(index,item) =>{
      let newBuild=[...this.state.currentBuild];
      let items=[...(newBuild[index].items)];
      items.push(item);
      newBuild[index]={...newBuild[index], items:items};
      this.setState({
        currentBuild: newBuild
      })
    },

    removeItemFromChamp:(index,indexOfItem) =>{
      let newBuild=[...this.state.currentBuild];
      let items=[...(newBuild[index].items)];
      items.splice(indexOfItem,1);
      newBuild[index]={...newBuild[index], items:items};
      this.setState({
        currentBuild: newBuild
      })
    },

    switchItemOnChamp:(index,item,itemIndex)=>{
      let newBuild=[...this.state.currentBuild];
      let items=[...(newBuild[index].items)];
      items.splice(itemIndex,1,item);
      newBuild[index]={...newBuild[index], items:items};
      this.setState({
        currentBuild: newBuild
      })
    },

    removeChampFromCurrentBuild:(index)=>{
      let newBuild=[...this.state.currentBuild];
      newBuild.splice(index,1);
      this.setState({
        currentBuild: newBuild
      })
    },

    addChampToCurrentBuild:(champ)=>{
      let curtChamp={
        id:champ.id,
        stars:champ.stars? champ.stars: 1,
        items:champ.items? champ.items: [],
      };/* 
      curtChamp.stars=1;
      curtChamp.items=[]; */
      this.setState({
        currentBuild: [...this.state.currentBuild,curtChamp]
      })
    },

    setCurrentBuild: (newBuild) => {
      this.setState({ currentBuild: newBuild })
    },

    setChampions: champions => {
      this.setState({ champions })
    },

    setItems: items => {
      this.setState({ items })
    },

    setTraits: traits => {
      this.setState({ traits })
    },

    setError: error => {
      console.error(error)
      this.setState({ error })
    },

    clearError: () => {
      this.setState({ error: null })
    },

  };

  render() {
    /* const value = {
      error: this.state.error,
      setError: this.setError,
      clearError: this.clearError,
      setTraits: this.setTraits,
      setItems: this.setItems,
      setChampions: this.setChampions,
      items: this.items,
      traits: this.traits,
      champions: this.champions
    } */
    return (
      <TftContext.Provider value={this.state}>
        {this.props.children}
      </TftContext.Provider>
    )
  }
}