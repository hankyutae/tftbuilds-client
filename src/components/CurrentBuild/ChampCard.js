import React from 'react';
import TftContext from '../../contexts/TftContext';
import calcStats from '../../services/calculate-final-stats-service';
import ImgLink from '../../services/create-img-link-service';
import './ChampCard.css';
import handleAddItem from '../../services/handle-add-item-service';
import ValidationError from '../ValidationError/ValidationError';
import SingleStat from './SingleStat';
import ChampCardUltimate from './ChampCardUltimate';
import CompleteItemInfoBox from './CompleteItemInfoBox';

class ChampCard extends React.Component {
  /* state = {
    isEditting: false,
    selectedItem: 1
  } */
  /* handleEditClick = () => {
    this.setState({
      isEditting: !this.state.isEditting
    })
  } */
  static contextType = TftContext

  handleStarsChange = (stars) => {
    this.context.changeStarsOnChamp(this.props.index, stars);
  }
  raiseStarsOnChamp = (e) => {
    e.stopPropagation();
    this.context.raiseStarsOnChamp(this.props.index);
  }
  lowerStarsOnChamp = (e) => {
    e.stopPropagation();
    this.context.lowerStarsOnChamp(this.props.index);
  }
  handleItemSelectionChange = (e) => {
    this.setState({
      selectedItem: Number(e.target.value),
    })
  }
  handleAddItemSubmit = (item) => {
    let res = handleAddItem(this.context.items, this.context.currentBuild[this.props.index].items, item);
    if (res === 'Too many items') {
      return;
    }
    else if (res[0] === 'switch') {
      this.context.switchItemOnChamp(this.props.index, res[1], res[2]);
    }
    else if (res[0] === 'add') {
      this.context.addItemToChamp(this.props.index, res[1]);
    }

  }
  handleRemoveItem = (itemIndex) => {
    this.context.removeItemFromChamp(this.props.index, Number(itemIndex));
  }

  // returns Array of <span> with stars of the right color.
  generateStars = (stars) => {
    let starColor = this.context.arrayOfStars[stars - 1];
    let starGen = [];
    for (let i = 0; i < stars; i++) {
      starGen.push(<span key={i} className={`fa fa-star ${starColor}`} ></span>);
    }
    return starGen
  }
  validateAddItem = () => {
    if (handleAddItem(this.context.items, this.context.currentBuild[this.props.index].items, 11) === 'Too many items') {
      return 'Potentially too many items, may remove items by clicking on the item icons below'
    }
  }
  createItemSelection = () => {
    /* let array=[];
    for(let i=0;i<this.context.numOfBasicItems;i++){
      array.push(<option key={i} value={`${i+1}`}> {this.context.items[i+1].name}</option>);
    } */
    let arr = [];
    for (let i = 1; i <= this.context.numOfBasicItems; i++) {

      arr.push(<img key={i} className='build-item-icon' src={ImgLink.createLink(this.context.items[i].icon)} alt={`${this.context.items[i].name} icon`} onClick={(e) => { e.stopPropagation(); this.handleAddItemSubmit(i); }} />)

    }
    for (let i = 1; i <= this.context.numOfBasicItems; i++) {
      for (let j = i; j <= this.context.numOfBasicItems; j++) {
        arr.push(<img key={i * 10 + j} className='build-item-icon' src={ImgLink.createLink(this.context.items[i * 10 + j].icon)} alt={`${this.context.items[i * 10 + j].name} icon`} onClick={(e) => { e.stopPropagation(); this.handleAddItemSubmit(i * 10 + j); }} />)
      }

    }
    return arr;
  }
  createItemSelectionGrid = () => {
    const completeGrid = [];
    const firstRow = [<td key={0}></td>];
    for (let i = 1; i <= this.context.numOfBasicItems; i++) {
      firstRow.push(<td key={i}><img className='grid-build-item-icon button-cursor' src={ImgLink.createLink(this.context.items[i].icon)} alt={`${this.context.items[i].name} icon`} onClick={(e) => { e.stopPropagation(); this.handleAddItemSubmit(i); }} /> </td>)

    }
    completeGrid.push(
      <tr key={0}>
        {firstRow}
      </tr>
    );

    for (let i = 1; i <= this.context.numOfBasicItems; i++) {
      let row = [];
      row.push(<td key={i}><img className='grid-build-item-icon button-cursor' src={ImgLink.createLink(this.context.items[i].icon)} alt={`${this.context.items[i].name} icon`} onClick={(e) => { e.stopPropagation(); this.handleAddItemSubmit(i); }} /> </td>);
      for (let j = 1; j <= this.context.numOfBasicItems; j++) {
        let index = (i < j) ? i * 10 + j : j * 10 + i;
        row.push(<td key={index}><img className='grid-build-item-icon button-cursor' src={ImgLink.createLink(this.context.items[index].icon)} alt={`${this.context.items[index].name} icon`} onClick={(e) => { e.stopPropagation(); this.handleAddItemSubmit(index); }} /> </td>);
      }
      completeGrid.push(
        <tr key={i}>
          {row}
        </tr>
      );
    }
    return completeGrid;


  }
  createStarsSelection = () => {
    let arr = [];
    this.context.arrayOfStars.forEach((star, index) => {
      arr.push(
        <div key={index} className="selecting-stars" onClick={(e) => {
          this.handleStarsChange(index + 1);
        }}>
          {this.generateStars(index + 1)}
        </div>
      )

    })
    return arr;
  }

  createStatTags = () => {
    return [
      "hp",
      "mana",
      "initalMana",
      "damage",
      "attackSpeed",
      "range",
      "critMultiplier",
      "critChance",
      "armor",
      "magicResist"
    ]
  }

  getStatToText = (statType) => {
    let table = {
      hp: "HP",
      mana: "Mana",
      damage: 'Damage',
      initalMana: "Init. Mana",
      magicResist: "MR",
      critChance: "Crit. Chance",
      critMultiplier: "Crit. Mult.",
      attackSpeed: "AS",
      range: "Range",
      armor: 'Armor',
    }
    return table[statType];
  }

  handleRemoveChamp = () => {
    this.context.removeChampFromCurrentBuild(this.props.index);
  }

  render() {
    const currentChamp = this.context.currentBuild[this.props.index];
    const currentChampInfo = this.context.champions[currentChamp.id]
    let statMod = calcStats({ ...currentChamp, ...currentChampInfo }, this.props.syn, this.context.items);

    return (
      <div className='build-display-champion-card'>

        <div className='champion-simple-flex'>
          {(this.props.screenSize!==2) &&
            <div className="edit-notice button-cursor button-hover-darken" onClick={() => this.context.toggleUltOnCurrentBuild(this.props.index)}>
              <strong>{currentChamp.isUltExpanded ? 'Hide Ult' : 'Show Ult'}</strong>
           </div>
          }
          <div className='b1'>
            <div className={`build-champ-icon-box `}>
              <img className={`build-champ-icon cost${currentChampInfo.cost}`} src={ImgLink.createLink(currentChampInfo.splash)} alt={currentChampInfo.name + ' icon'} />
              <div className={`build-champ-icon-box-price${currentChampInfo.cost}`}>
                {currentChampInfo.cost}
              </div>
            </div>
            <p>
              {currentChampInfo.name}
            </p>
          </div>
          <div className='b2'>
            {currentChampInfo.traits.map((trait, index) =>
              <p key={index}>{trait}</p>
            )}
          </div>
          <div className='b3'>
            <i className="fa fa-chevron-up button-cursor increase-fontsize-on-hover" aria-hidden="true" onClick={this.raiseStarsOnChamp}></i>
            <div className="champ-stars">
              {this.generateStars(currentChamp.stars)}
            </div>

            <span className="fa fa-chevron-down button-cursor increase-fontsize-on-hover" aria-hidden="true" onClick={this.lowerStarsOnChamp}></span>
          </div>
          <div className='b4'>
            <div className='button-cursor build-buttons button-hover-darken add-items-button' onClick={(e) => { e.stopPropagation(); this.context.toggleAddItemOnCurrentBuild(this.props.index) }}>
              {currentChamp.isAddItemExpanded ? "Hide Items" : "Add Items"}
            </div>
            <div className='build-items-box'>
              {
                currentChamp.items.map((item, index) => (
                  <div key={index} className='item-img-div' >
                    <img className='build-item-icon button-cursor' src={ImgLink.createLink(this.context.items[item].icon)} alt={`${this.context.items[item].name} icon`} onClick={(e) => { e.stopPropagation(); this.handleRemoveItem(index); }} />
                  </div>
                )
                )
              }
            </div>

            {this.props.screenSize===0 &&
              <div className='button-cursor build-buttons button-hover-darken' onClick={(e) => { e.stopPropagation(); this.context.toggleStatsOnCurrentBuild(this.props.index) }}>
                {currentChamp.isStatsExpanded ? 'Hide Stats' : 'Show Stats'}
              </div>
              /* :
              <div className='button-cursor' onClick={(e) => { e.stopPropagation(); this.context.toggleUltOnCurrentBuild(this.props.index) }}>Show Ult Stats</div> */
            }

          </div>

          {!(this.props.screenSize===0) &&
            <div className='build-stats-table-div'>
              <table className="build-stats-table">
                <tbody>
                  <tr>
                    {
                      this.createStatTags().map((statType,index) => {
                        return (
                          <td key={index} className={`table-stat-type ${statType}`}>
                            {this.getStatToText(statType)}
                          </td>
                        );
                      })
                    }
                  </tr>
                  <tr>
                    {
                      this.createStatTags().map((statType) => {
                        return <SingleStat key={statType} champ={currentChamp} champInfo={currentChampInfo} statMod={statMod} statType={statType} />
                      })
                    }
                  </tr>
                </tbody>
              </table>
            </div>
          }

          {(this.props.screenSize===2) &&
            <ChampCardUltimate champ={currentChamp} champInfo={currentChampInfo} statMod={statMod} />
          }

          <div className='b5-remove button-cursor button-hover-darken' onClick={(e) => {
            e.stopPropagation();
            this.context.removeChampFromCurrentBuild(this.props.index);
          }}>
            {/* <div>R</div>
            <div>E</div>
            <div>M</div>
            <div>O</div>
            <div>V</div>
            <div>E</div> */}
            <div>E</div>
            <div>R</div>
            <div>A</div>
            <div>S</div>
            <div>E</div>
          </div>
        </div>
        <div className="build-display-card-expansion">
          {this.props.screenSize===0 && currentChamp.isStatsExpanded &&
            <div className='build-stats-table-div'>
              <table className="build-stats-table">
                <tbody>
                  <tr>
                    {
                      this.createStatTags().map((statType,index) => {
                        return (
                          <td key={index} className={`table-stat-type ${statType}`}>
                            {this.getStatToText(statType)}
                          </td>
                        );
                      })
                    }
                  </tr>
                  <tr>
                    {
                      this.createStatTags().map((statType) => {
                        return <SingleStat key={statType} champ={currentChamp} champInfo={currentChampInfo} statMod={statMod} statType={statType} />
                      })
                    }
                  </tr>
                </tbody>
              </table>
            </div>

          }
          {(this.props.screenSize!==2) && currentChamp.isUltExpanded &&
            <ChampCardUltimate champ={currentChamp} champInfo={currentChampInfo} statMod={statMod} />
          }
            
          

          {/* <section className='build-display-change-stars'>
              <div>Choose Stars:</div>
              <div className='selecting-stars-box'>
                {this.createStarsSelection()}
              </div>
            </section> */}

          {


          }
          {currentChamp.isAddItemExpanded &&
            <div>
              <section className='build-display-add-items'>
                <div>Click on icon to add:  </div>
                <div className='build-display-item-grid-box'>
                  <table className='build-display-item-grid'>
                    <tbody>
                      {this.createItemSelectionGrid()}
                    </tbody>
                  </table>
                </div>
                <ValidationError message={this.validateAddItem()} />
              </section>
              {currentChamp.items.length !== 0 &&
                <section className='build-display-expanded-show-items'>
                  {currentChamp.items.map((item, index) => {
                    return <CompleteItemInfoBox key={index} items={this.context.items} item={item} isExpanded={true} itemIndex={index} handleRemove={this.handleRemoveItem} />
                  })}
                </section>
              }

            </div>}
        </div>



      </div>
    );
  }

}

export default ChampCard;