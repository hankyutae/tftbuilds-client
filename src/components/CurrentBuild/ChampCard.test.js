import React from 'react';
import ReactDOM from 'react-dom';
import ChampCard from './ChampCard';
import {BrowserRouter} from 'react-router-dom';
import {TftProvider} from '../../contexts/TftContext';
import champions from '../TestHelpers/champions';
import items from '../TestHelpers/items';
import traits from '../TestHelpers/traits';
import renderer from "react-test-renderer";


let champ=champions[(Object.keys(champions)[0])];
let syn={};
let currentBuild= [ 
  {
    id:Object.keys(champions)[0],
    items: [2],
    stars: 2
  }
];

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <TftProvider champions={champions} items={items} traits={traits} currentBuild={currentBuild}>
      <BrowserRouter>
        <ChampCard champ={champ} syn={syn} index={0} screensize={2}/>
      </BrowserRouter>
    </TftProvider>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("renders the UI as expected", () => {
  const tree = renderer
    .create(
      <TftProvider champions={champions} items={items} traits={traits} currentBuild={currentBuild}>
        <BrowserRouter>
          <ChampCard champ={champ} syn={syn} index={0} screensize={2}/>
        </BrowserRouter>
      </TftProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
