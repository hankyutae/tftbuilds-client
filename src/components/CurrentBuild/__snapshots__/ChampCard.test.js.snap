// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="build-display-champion-card"
>
  <div
    className="champion-simple-flex"
  >
    <div
      className="edit-notice button-cursor button-hover-darken"
      onClick={[Function]}
    >
      <strong>
        Show Ult
      </strong>
    </div>
    <div
      className="b1"
    >
      <div
        className="build-champ-icon-box "
      >
        <img
          alt="Twisted Fate icon"
          className="build-champ-icon cost2"
          src="http://raw.communitydragon.org/latest/game/assets/ux/tft/championsplashes/tft_twistedfate.png"
        />
        <div
          className="build-champ-icon-box-price2"
        >
          2
        </div>
      </div>
      <p>
        Twisted Fate
      </p>
    </div>
    <div
      className="b2"
    >
      <p>
        Pirate
      </p>
      <p>
        Sorcerer
      </p>
    </div>
    <div
      className="b3"
    >
      <i
        aria-hidden="true"
        className="fa fa-chevron-up button-cursor increase-fontsize-on-hover"
        onClick={[Function]}
      />
      <div
        className="champ-stars"
      >
        <span
          className="fa fa-star silver"
        />
        <span
          className="fa fa-star silver"
        />
      </div>
      <span
        aria-hidden="true"
        className="fa fa-chevron-down button-cursor increase-fontsize-on-hover"
        onClick={[Function]}
      />
    </div>
    <div
      className="b4"
    >
      <div
        className="button-cursor build-buttons button-hover-darken add-items-button"
        onClick={[Function]}
      >
        Add Items
      </div>
      <div
        className="build-items-box"
      >
        <div
          className="item-img-div"
        >
          <img
            alt="Recurve Bow icon"
            className="build-item-icon button-cursor"
            onClick={[Function]}
            src="http://raw.communitydragon.org/latest/game/assets/maps/particles/tft/tft_item_recurvebow.tft.png"
          />
        </div>
      </div>
    </div>
    <div
      className="build-stats-table-div"
    >
      <table
        className="build-stats-table"
      >
        <tbody>
          <tr>
            <td
              className="table-stat-type hp"
            >
              HP
            </td>
            <td
              className="table-stat-type mana"
            >
              Mana
            </td>
            <td
              className="table-stat-type initalMana"
            >
              Init. Mana
            </td>
            <td
              className="table-stat-type damage"
            >
              Damage
            </td>
            <td
              className="table-stat-type attackSpeed"
            >
              AS
            </td>
            <td
              className="table-stat-type range"
            >
              Range
            </td>
            <td
              className="table-stat-type critMultiplier"
            >
              Crit. Mult.
            </td>
            <td
              className="table-stat-type critChance"
            >
              Crit. Chance
            </td>
            <td
              className="table-stat-type dodgeChance"
            >
              Dodge
            </td>
            <td
              className="table-stat-type armor"
            >
              Armor
            </td>
            <td
              className="table-stat-type magicResist"
            >
              MR
            </td>
          </tr>
          <tr>
            <td
              className="hp"
              onClick={[Function]}
            >
              <strong>
                900
              </strong>
            </td>
            <td
              className="mana"
              onClick={[Function]}
            >
              <strong>
                50
              </strong>
            </td>
            <td
              className="initalMana"
              onClick={[Function]}
            >
              <strong>
                0
              </strong>
            </td>
            <td
              className="damage"
              onClick={[Function]}
            >
              <strong>
                72
              </strong>
            </td>
            <td
              className="button-cursor button-hover-darken attackSpeed"
              onClick={[Function]}
            >
              <strong>
                0.84
              </strong>
            </td>
            <td
              className="range"
              onClick={[Function]}
            >
              <strong>
                3
              </strong>
            </td>
            <td
              className="critMultiplier"
              onClick={[Function]}
            >
              <strong>
                1.5
              </strong>
            </td>
            <td
              className="critChance"
              onClick={[Function]}
            >
              <strong>
                0.25
              </strong>
            </td>
            <td
              className="dodgeChance"
              onClick={[Function]}
            >
              <strong>
                0
              </strong>
            </td>
            <td
              className="armor"
              onClick={[Function]}
            >
              <strong>
                20
              </strong>
            </td>
            <td
              className="magicResist"
              onClick={[Function]}
            >
              <strong>
                20
              </strong>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      className="b5-remove button-cursor button-hover-darken"
      onClick={[Function]}
    >
      <div>
        E
      </div>
      <div>
        R
      </div>
      <div>
        A
      </div>
      <div>
        S
      </div>
      <div>
        E
      </div>
    </div>
  </div>
  <div
    className="build-display-card-expansion"
  />
</div>
`;
